src/condition.cpp

```diff
@@ -1039,15 +1040,15 @@ void ConditionSoul::serialize(PropWriteStream& propWriteStream)
 	propWriteStream.write<uint32_t>(soulTicks);
 }
 
-bool ConditionSoul::executeCondition(Creature* creature, int32_t interval)
+bool ConditionSoul::executeCondition(std::shared_ptr<Creature> creature, int32_t interval)
 {
 	if (!creature) {
 		return false;
 	}
 
 	internalSoulTicks += interval;
 
-	if (Player* player = creature->getPlayer()) {
+	if (auto player = creature->getPlayer()) {
```

Comment:
Could this be `const auto`

```suggestion
	if (const auto player = creature->getPlayer()) {
```